### PR TITLE
Download TX history for exporting on-demand

### PR DESCRIPTION
**Pull request by neuodev**
https://emurgo.atlassian.net/browse/YOEXT-482

Based on #3055

### CODE REVIEW

**Comment by neuodev** [comment]
[autoscript] ⚠️ `Ahmed/tx-export-on-demand` doesn't full our branch naming convention.
Should follow `<tag>/<jira-ticket-id>/<short-name>`
<details>
<summary>Details</summary>

- `<tag>`: `fix`, `feat`, `chore`, `refactor`, `test`, ...
- `<jira-ticket-id>`: The issue ID of JIRA all capital letters
- `<short-name>`: A short and descriptive name for the branch
</details>

[](__JIRA_COMMENT_FLAG)

File: packages/yoroi-extension/app/api/ada/lib/storage/bridge/traitUtils.js [hunk 1]
```diff
@@ -166,6 +167,42 @@ export async function getAllAddressesForDisplay(
...
 }
 
+export async function getAllAddressesForWallet(
+  publicDeriver: PublicDeriver<>,
+): Promise<{|
+  utxoAddresses: Array<$ReadOnly<AddressRow>>,
+  accountingAddresses: Array<$ReadOnly<AddressRow>>,
+|}> {
+  const p = asHasLevels<ConceptualWallet>(publicDeriver);
+    if (!p) {
```
Comment (yushih) [inline packages/yoroi-extension/app/api/ada/lib/storage/bridge/traitUtils.js]: @AhmedIbrahim336 please mind the indentations. Thanks.

**Review by zuzunker — APPROVED** [review]

**Review by zuzunker — COMMENTED** [review]
@Nebyt , plz /check , need to make sure all the backend APIs are still compatible

**Review by zuzunker — COMMENTED** [review]
/check

**Pull request merged**